vm/src/macros.rs

```diff
@@ -83,14 +83,20 @@ macro_rules! arg_check {
         //  check if the type matches. If not, return with error
         //  assign the arg to a variable
         $(
-            let $optional_arg_name = if arg_count < $args.args.len() {
-                type_check!($vm, $args, arg_count, $optional_arg_name, $optional_arg_type);
-                let ret = Some(&$args.args[arg_count]);
-                #[allow(unused_assignments)]
-                {
-                    arg_count += 1;
+            let len = $args.args.len();
```

Comment:
Wait, this change looks really weird.
Builtin functions can have optional arguments that are not keyword arguments. They way you did it, you seem to assume that optional arguments can *only* be kwargs. This breaks it completely.
(that said, admittably handling optional arguments that can be either positional or keyword is a bit awkward in the current model)
Also, it seems to assume that kwargs are passed in the same order as they were declared?
(Also, shouldn't this change be in a separate PR?)

Reply:
I reverted the changes